tests/testthat/test-parameters.R

```diff
@@ -0,0 +1,403 @@
+  withr::with_tempdir({
+    write_sample_parameters_file(
+      value = expected,
+      path = path,
+      state = "test",
+      disease = disease,
+      parameter = parameter,
+      param = parameter,
+      start_date = start_date,
+      end_date = NA
+    )
+    actual <- read_interval_pmf(
+      path = path,
+      parameter = parameter,
+      disease = disease,
+      as_of_date = start_date + 1,
+      state = "test"
+    )
+  })
+  expect_equal(actual, expected)
+})
+
+test_that("Invalid PMF errors", {
+  expected <- c(0.8, -0.1)
+  path <- "test.parquet"
+  parameter <- "right_truncation"
+  start_date <- as.Date("2023-01-01")
+
+  # COVID-19
+  disease <- "COVID-19"
+  withr::with_tempdir({
+    write_sample_parameters_file(
+      value = expected,
+      path = path,
+      state = "test",
+      disease = disease,
+      parameter = parameter,
+      param = parameter,
+      start_date = start_date,
+      end_date = NA
+    )
+    expect_error(read_interval_pmf(
+      path = path,
+      parameter = parameter,
+      disease = disease,
+      as_of_date = start_date + 1,
+      state = "test"
+    ))
+  })
+})
+
+
+test_that("Can read delay on happy path", {
+  expected <- c(0.8, 0.2)
+  path <- "test.parquet"
+  parameter <- "delay"
+  start_date <- as.Date("2023-01-01")
+
+  # COVID-19
+  disease <- "COVID-19"
+  withr::with_tempdir({
+    write_sample_parameters_file(
+      value = expected,
+      path = path,
+      state = NA,
+      disease = disease,
+      parameter = parameter,
+      param = parameter,
+      start_date = start_date,
+      end_date = NA
+    )
+    actual <- read_interval_pmf(
+      path = path,
+      disease = disease,
+      as_of_date = start_date + 1,
+      parameter = parameter
+    )
+  })
+  expect_equal(actual, expected)
+
+
+  # Influenza
+  disease <- "Influenza"
+  withr::with_tempdir({
+    write_sample_parameters_file(
+      value = expected,
+      path = path,
+      state = NA,
+      disease = disease,
+      parameter = parameter,
+      param = parameter,
+      start_date = start_date,
+      end_date = NA
+    )
+    actual <- read_interval_pmf(
+      path = path,
+      disease = disease,
+      as_of_date = start_date + 1,
+      parameter = parameter
+    )
+  })
+  expect_equal(actual, expected)
+})
+
+
+test_that("Not a PMF errors", {
+  expected <- "hello"
+  path <- "test.parquet"
+  parameter <- "delay"
+  start_date <- as.Date("2023-01-01")
+
+  # COVID-19
+  disease <- "COVID-19"
+  withr::with_tempdir({
+    write_sample_parameters_file(
+      value = expected,
+      path = path,
+      state = NA,
+      disease = disease,
+      parameter = parameter,
+      param = parameter,
+      start_date = start_date,
+      end_date = NA
+    )
+    expect_error(read_interval_pmf(
+      path = path,
+      disease = disease,
+      as_of_date = start_date + 1,
+      parameter = parameter
+    ))
+  })
+})
+
+test_that("Invalid disease errors", {
```

Comment:
Perhaps include more detail on the test names here? e.g. invalid disease errors for what function?

Reply:
I don't think that adds clarity. When the tests fail, they look like this:

![image](https://github.com/user-attachments/assets/33b5d356-ad93-47c8-9a4a-fbb6b0d3985e)

So the test name and the file are listed, with the exact call also displayed. I think it's better to have a short, descriptive slug to give a sense for what kind of thing is going wrong in the call. If you still think a change is needed, please leave a suggestion!